sklearn/preprocessing/data.py

```diff
@@ -3046,28 +3064,28 @@ def _transform(self, X):
                              " Expected %d, got %d."
                              % (indices.shape[0] - 1, n_features))
 
-        # We use only those ordinal features of X that are known using fit.
-        # i.e lesser than n_values_ using mask.
-        # This means, if self.handle_unknown is "ignore", the row_indices and
-        # col_indices corresponding to the unknown ordinal feature are
-        # ignored.
+        # We clip those ordinal features of X that are greater than n_values_
+        # using mask.
+        # This means, if self.handle_greater is "ignore", the row_indices and
+        # col_indices corresponding to the greater ordinal feature are all
+        # filled with ones.
         mask = (X < self.n_values_).ravel()
         if np.any(~mask):
-            if self.handle_unknown not in ['error', 'ignore']:
-                raise ValueError("handle_unknown should be either 'error' or "
-                                 "'ignore' got %s" % self.handle_unknown)
-            if self.handle_unknown == 'error':
+            if self.handle_greater not in ['error', 'clip']:
+                raise ValueError("handle_greater should be either 'error' or "
+                                 "'clip' got %s" % self.handle_greater)
+            if self.handle_greater == 'error':
                 raise ValueError("unknown ordinal feature present %s "
                                  "during transform." % X.ravel()[~mask])
 
-        column_start = np.tile(indices[:-1], n_samples)[mask]
-        column_end = (X + indices[:-1]).ravel()[mask]
+        X_ceil = np.where(mask.reshape(X.shape), X, self.n_values_ - 1)
```

Comment:
Now is the transformation logic identical in the fit and transform cases? Are we now able to define fit_transform as just fit + transform?

Reply:
What I mean is we should be able to refactor this.